itm/dns_apps.go

```diff
@@ -107,16 +121,13 @@ func (s *dnsAppsServiceImpl) Update(id int, opts *DNSAppOpts, publish bool) (*DN
 			publishVal,
 		},
 	}
-	resp, err := s.client.put(getDNSAppPath(id), jsonOpts, qs)
+	resp, err := s.client.put(getDNSAppPath(id), opts.toJSON(), qs)
 	if err != nil {
-		log.Printf("Error issuing put request from DNSAppsServiceImpl.Update: %v", err)
+		log.Printf("[WARN] Error issuing put request from DNSAppsServiceImpl.Update: %v", err)
 		return nil, err
 	}
 	if 200 != resp.StatusCode {
-		return nil, &UnexpectedHTTPStatusError{
-			Expected: 200,
-			Got:      resp.StatusCode,
-		}
+		return nil, fmt.Errorf("Unexpected HTTP response. HTTP status code: %d. Body: %s", resp.StatusCode, resp.Body)
```

Comment:
Does this line need an error level like the other errors? (like line 106?)

Reply:
Good eye! Actually it's the other way around. The convention is to include the log level in calls to `log.Printf` and `log.Println`, not in the strings used to create error objects, so I will remove these.